src/include/catalog/catalog_entry/table_catalog_entry.h

```diff
@@ -58,6 +59,7 @@ class KUZU_API TableCatalogEntry : public CatalogEntry {
     const binder::PropertyDefinition& getProperty(const std::string& propertyName) const;
     const binder::PropertyDefinition& getProperty(common::idx_t idx) const;
     virtual common::column_id_t getColumnID(const std::string& propertyName) const;
+    common::column_id_t getColumnID(common::idx_t idx) const;
```

Comment:
Remove this. Always get through propertyName instead of idx because there will be internal properties causing index to shift.

Reply:
@acquamarin, is passing a property name to prepareScan acceptable? You'd originally said to pass a column id, I'd changed it to a property index (since you can look up properties to get their type by index and name, but not by column id, and can't look up the index or name using the column ID) as I'd assumed that would be preferable than passing names around.

Unfortunately it seems I then mixed up the types in the signatures, so it's currently using property indices internally, but the type and argument names in prepareScan indicate that it expects a column ID (unfortunately `column_id_t` and `idx_t` are aliases of the same type, so the compiler won't catch that).